antarest/study/service.py

```diff
@@ -1411,32 +1411,63 @@ def _create_edit_study_command(
                 )
         raise NotImplementedError()
 
-    def _edit_study_using_command(self, study: Study, url: str, data: SUB_JSON) -> ICommand:
+    def _edit_study_using_command(
+        self,
+        study: Study,
+        url: str,
+        data: SUB_JSON,
+        *,
+        create_missing: bool = False,
+    ) -> ICommand:
         """
-        Replace data on disk with new, using ICommand
+        Replace data on disk with new, using variant commands.
+
+        In addition to regular configuration changes, this function also allows the end user
+        to store files on disk, in the "user" directory of the study (without using variant commands).
+
         Args:
             study: study
             url: data path to reach
             data: new data to replace
+            create_missing: Flag to indicate whether to create file or parent directories if missing.
         """
-
         study_service = self.storage_service.get_storage(study)
         file_study = study_service.get_raw(metadata=study)
-        tree_node = file_study.tree.get_node(url.split("/"))
+
+        file_relpath = PurePosixPath(url.strip().strip("/"))
+        file_path = study_service.get_study_path(study).joinpath(file_relpath)
+        create_missing &= not file_path.exists()
+        if create_missing:
+            # IMPORTANT: We prohibit deep file system changes in private directories.
+            # - File and directory creation is only possible for the "user" directory,
+            #   because the "input" and "output" directories are managed by Antares.
+            # - We also prohibit writing files in the "user/expansion" folder which currently
+            #   contains the Xpansion tool configuration.
+            #   This configuration should be moved to the "input/expansion" directory in the future.
+            if file_relpath and file_relpath.parts[0] == "user" and file_relpath.parts[1] != "expansion":
+                # In the case of variants, we must write the file directly in the study's snapshot folder,
+                # because the "user" folder is not managed by the command mechanism.
```

Comment:
Indeed, so it's temporary and not the best way to do it. The command `update_raw_file` does what we want but it's a bit deprecated.

Reply:
What do you mean by "temporary"?

The user can upload a file to create the target file if he set `create_missing=True`, so it works for a missing file in that case only.

Reply:
What I mean is that you will directly write in the snapshot which is by desing temporary.
The right way to do it would be to create a command because when the snapshot will be cleaned, this file will disappear forever.
But as I wrote i think the `update_raw_file` command does not work for unexisting files.
So this solution is not optimal but i do not see any workaround